src/chatd.cpp

```diff
@@ -4375,6 +4500,23 @@ void Chat::deleteMessagesBefore(Idx idx)
     }
 }
 
+void Chat::truncateByRetentionTime(Idx idx)
+{
+    if (idx >= mForwardStart)
+    {
+        mBackwardList.clear();
+        auto delCount = idx - mForwardStart;
+        auto end = mForwardList.begin() + delCount;
+        mForwardList.erase(mForwardList.begin(), end);
+        mForwardList.erase(mForwardList.begin());
```

Comment:
Isn't it the same than... ??
```suggestion
        mForwardList.erase(mForwardList.begin(), end + 1);
```